tools/windows-node-installer/pkg/cloudprovider/factory.go

```diff
@@ -13,7 +16,33 @@ import (
 // this function requires specifying the credentialAccountID of the user's credential account.
 // The resourceTrackerDir is where the `windows-node-installer.json` file which contains IDs of created instance and
 // security group will be created.
-func CloudProviderFactory(kubeconfigPath, credentialPath, credentialAccountID, resourceTrackerDir string) (Cloud, error) {
+
+//struct to hold azure specific information.
+type svc struct {
+	resourceGroupName     string // name of the resource group
+	resourceGroupLocation string // name of the resource location
+
+	deploymentName string                 // name of the deployment
+	templateFile   map[string]interface{} // name of the templateFile (need from user)
+	parametersFile map[string]interface{} // name of the parametersFile (need from user)
+	authdata       map[string]interface{} // Azure auth data
+}
```

Comment:
Please add the comments on a separate line. Moreover it is obvious that `resourceGroupName` is the name of the resource group. You need to explain what they actually mean from an Azure POV.

Reply:
will try to add more detail in it.